### PR TITLE
Fixed IndexError in constant order multiplication.

### LINKED ISSUE #9192: O(y + 1) = O(1)
I'm getting a very silly output from some simple big-O expressions involving O(1). I'm using Sympy 0.7.5, but I don't see anything in the release notes for 0.7.6 that addresses this issue.

```
    >>> y = Symbol('y')
    >>> O(y + 1)
    O(1)
    >>> x = O(1)
    >>> x * x
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.3/site-packages/sympy/core/decorators.py", line 77, in __sympifyit_wrapper
    return func(a, b)
  File "/usr/local/lib/python3.3/site-packages/sympy/core/decorators.py", line 118, in binary_op_wrapper
    return func(self, other)
  File "/usr/local/lib/python3.3/site-packages/sympy/core/expr.py", line 125, in __mul__
    return Mul(self, other)
  File "/usr/local/lib/python3.3/site-packages/sympy/core/cache.py", line 93, in wrapper
    r = func(*args, **kw_args)
  File "/usr/local/lib/python3.3/site-packages/sympy/core/operations.py", line 40, in __new__
    c_part, nc_part, order_symbols = cls.flatten(args)
  File "/usr/local/lib/python3.3/site-packages/sympy/core/mul.py", line 237, in flatten
    o, order_symbols = o.as_expr_variables(order_symbols)
  File "/usr/local/lib/python3.3/site-packages/sympy/series/order.py", line 284, in as_expr_variables
    if order_symbols[0][1] != self.point[0]:
IndexError: tuple index out of range
```
```
    >>> O(1) ** O(1) # should be O(1)
    O(1)**O(1) 
```
However, this one works, which is strange.
```
    >>> O(1) + y**y
    y**y + O(1)
```

### PR DESCRIPTION
**Pull request by debugger22**
Fixes #9192

### CODE REVIEW

**Comment by debugger22** [comment]
@smichr Do you want to take a look here?

**Comment by debugger22** [comment]
@jcrist Does this look good?

File: sympy/series/order.py [hunk 1]
```diff
@@ -304,7 +304,7 @@ def as_expr_variables(self, order_symbols):
                not all(p == self.point[0] for p in self.point):
                 raise NotImplementedError('Order at points other than 0 '
                     'or oo not supported, got %s as a point.' % point)
-            if order_symbols[0][1] != self.point[0]:
+            if len(order_symbols) > 0 and order_symbols[0][1] != self.point[0]:
```
Comment (jcrist) [inline sympy/series/order.py]: why not just `if ordered_symbols and ...`?

Comment (debugger22) [inline sympy/series/order.py]: Yes, that would be better.

File: sympy/series/tests/test_order.py [hunk 2]
```diff
@@ -405,3 +405,7 @@ def test_order_subs_limits():
 
     assert Order(x**2).subs(x, y - 1) == Order((y - 1)**2, (y, 1))
     assert Order(10*x**2, (x, 2)).subs(x, y - 1) == Order(1, (y, 3))
+
+
+def test_issue_9192():
+    assert O(1)*O(1) == O(1)
```
Comment (jcrist) [inline sympy/series/tests/test_order.py]: Does this also fix the other issue? If so, you should add a test for it as well:

```
>>> O(1) ** O(1) # should be O(1)
O(1)**O(1) 
```

Comment (debugger22) [inline sympy/series/tests/test_order.py]: No, it doesn't. We can't close that issue even after merging it.

Comment (jcrist) [inline sympy/series/tests/test_order.py]: Should that be part of this PR? As thetest title is named after the issue, it should probably test all the facets of the issue (at least if they're related). In this case they don't appear to be, so perhaps change the test title?

Comment (debugger22) [inline sympy/series/tests/test_order.py]: Ok, fixed that and added test too.

**Comment by jcrist** [comment]
Looks good. Merging.

**Pull request merged**